### PR TITLE
feat(@aws-amplify/amazon-cognito-identity-js): for signUp, returns CodeDeliveryDetails

### PR DESCRIPTION
**Pull request by aaronseatsmart**
*Issue #, if available:*
N/A
*Description of changes:*
Adds CodeDeliveryDetails to returnData when calling userPool.signUp(). Need this info to label the verification code form that is displayed to user after signUp. The way it is done with hosted-ui.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by aaronseatsmart** [comment]
Any update on this one-liner? Let me know if I missed something about how to do a PR, I tried to follow repos guidelines.

**Comment by aaronseatsmart** [comment]
Anybody home? Is there anything additional I need to do to get this PR in?

File: packages/amazon-cognito-identity-js/src/CognitoUserPool.js [hunk 1]
```diff
@@ -110,6 +110,7 @@ export default class CognitoUserPool {
         user: new CognitoUser(cognitoUser),
         userConfirmed: data.UserConfirmed,
         userSub: data.UserSub,
+        CodeDeliveryDetails: data.CodeDeliveryDetails,
```
Comment (powerful23) [inline packages/amazon-cognito-identity-js/src/CognitoUserPool.js]: Can you change it to `codeDeliveryDetails: data.CodeDeliveryDetails,`?

**Comment by powerful23** [comment]
@aaronseatsmart besides this change, the typing declaration file should also be changed accordingly: https://github.com/aws-amplify/amplify-js/blob/master/packages/amazon-cognito-identity-js/index.d.ts#L137 I can make those changes in another pr if necessary. Please respond if you are willing to continue working on this pr.

**Pull request merged**